lib/travis/logs/app.rb

```diff
@@ -126,6 +127,33 @@ def initialize(auth_token: ENV['AUTH_TOKEN'].to_s,
         body nil
       end
 
+      get '/log-parts/:job_id' do
+        halt 500, 'authentication token is not set' if auth_token.empty?
+        halt 403 unless authorized?(request)
+
+        job_id = Integer(params[:job_id])
+        part_numbers = if params[:part_numbers].nil?
```

Comment:
I'm generally 👎 to compact ternaries like that.  I realize that this makes me a questionable Rubby programber.